checkov/arm/checks/resource/StorageAccountsTransportEncryption.py

```diff
@@ -1,25 +1,30 @@
+from __future__ import annotations
+
+from typing import Any
+
 from checkov.arm.base_resource_check import BaseResourceCheck
 from checkov.common.models.enums import CheckResult, CheckCategories
 from checkov.common.util.type_forcers import force_int
 
 
 class StorageAccountsTransportEncryption(BaseResourceCheck):
-    def __init__(self):
+    def __init__(self) -> None:
         # supportsHttpsTrafficOnly: Allows https traffic only to storage service if sets to true. The default value is
         # true since API version 2019-04-01.
         name = "Ensure that 'supportsHttpsTrafficOnly' is set to 'true'"
         id = "CKV_AZURE_3"
-        supported_resources = ['Microsoft.Storage/storageAccounts']
-        categories = [CheckCategories.ENCRYPTION]
+        supported_resources = ("Microsoft.Storage/storageAccounts",)
+        categories = (CheckCategories.ENCRYPTION,)
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
-    def scan_resource_conf(self, conf):
-        if "properties" in conf:
-            if "supportsHttpsTrafficOnly" in conf["properties"]:
-                if str(conf["properties"]["supportsHttpsTrafficOnly"]).lower() == "true":
-                    return CheckResult.PASSED
-                else:
-                    return CheckResult.FAILED
+    def scan_resource_conf(self, conf: dict[str, Any]) -> CheckResult:
+        properties = conf.get("properties")
+        if isinstance(properties, dict):
+            https = str(properties.get("supportsHttpsTrafficOnly")).lower()
+            if https == "true":
+                return CheckResult.PASSED
+            elif https == "false":
+                return CheckResult.FAILED
```

Comment:
```suggestion
            return CheckResult.PASSED if https == "true" else CheckResult.FAILED
```

Reply:
this was my original approach 😄 , but it breaks one use case. I think I have an idea on how to make it work 🤔